RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -510,18 +500,24 @@ class PumpOpsSynchronous {
                 timestamp.timeZone = pump.timeZone
 
                 if let date = timestamp.date?.addingTimeInterval(timeAdjustmentInterval) {
-                    if date.timeIntervalSince(startDate) < -eventTimestampDeltaAllowance {
-                        NSLog("Found event at (%@) to be more than %@s before startDate(%@)", date as NSDate, String(describing: eventTimestampDeltaAllowance), startDate as NSDate);
-                        return (events: events, hasMoreEvents: false)
-                    } else if date.timeIntervalSince(timeCursor) > eventTimestampDeltaAllowance {
-                        NSLog("Found event (%@) out of order in history. Ending history fetch.", date as NSDate)
-                        return (events: events, hasMoreEvents: false)
-                    } else {
-                        if (date.compare(startDate) != .orderedAscending) {
-                            timeCursor = date
-                        }
-                        events.insert(TimestampedHistoryEvent(pumpEvent: event, date: date), at: 0)
+                    
+                    let possibleCancelBecauseOfDate = !event.isDelayedAppend(withPumpModel: pumpModel)
+                    
+                    if possibleCancelBecauseOfDate {
+                        if date <= startDate {
+                            // Success, we have all the events we need
+                            //NSLog("Found event at or before startDate(%@)", date as NSDate, String(describing: eventTimestampDeltaAllowance), startDate as NSDate);
+                            return (events: events, hasMoreEvents: false, cancelledEarly: false)
+                        } else if date.timeIntervalSince(timeCursor) > TimeInterval(minutes: 60) {
+                            NSLog("Found event (%@) out of order in history. Ending history fetch.", date as NSDate)
+                            return (events: events, hasMoreEvents: false, cancelledEarly: true)
+                        }                    }
+                    
+                    if (date.compare(startDate) != .orderedAscending) {
```

Comment:
With swift 3, dates can be compared using > & <.  I think this can be re-written:

`if date >= startDate`

But I think we just want to update timeCursor for any non-delayed-append events.